src/impl/load.c

```diff
@@ -0,0 +1,289 @@
+                                                identifier));
+      }
+      return NULL;
+  }
+  
+  return gcg_content;
+}
+
+char *get_woogles_gcg_string(const char *identifier, ErrorStack *error_stack) {
+    char *game_id_str = NULL;
+
+    // Check if this is a Woogles URL
+    if (strstr(identifier, "woogles.io") != NULL) {
+        // Extract game ID from Woogles URL using regex
+        regex_t regex;
+        regmatch_t matches[2];
+
+        // Pattern matches /game/[alphanumeric]
+        int regex_status = regcomp(&regex, "/game/([a-zA-Z0-9]+)", REG_EXTENDED);
+        if (regex_status != 0) {
+            error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                            string_duplicate("Failed to compile regex for woogles URL parsing"));
+            return NULL;
+        }
+
+        int match_result = regexec(&regex, identifier, 2, matches, 0);
+        if (match_result == 0) {
+            int start = matches[1].rm_so;
+            int end = matches[1].rm_eo;
+            int id_len = end - start;
+            
+            game_id_str = malloc_or_die(id_len + 1);
+            strncpy(game_id_str, identifier + start, id_len);
+            game_id_str[id_len] = '\0';
+        } else {
+            regfree(&regex);
+            error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                            get_formatted_string("Failed to extract game ID from woogles URL: %s", identifier));
+            return NULL;
+        }
+
+        regfree(&regex);
+    } else {
+        // Check if it's a standalone woogles game ID (must be alphanumeric only)
+        if (is_all_digits_or_empty(identifier)) {
+            // All digits likely means cross-tables, not woogles
+            return NULL;
+        }
+        // Check if identifier is alphanumeric only
+        for (int i = 0; identifier[i]; i++) {
+            if (!isalnum(identifier[i])) {
+                // Contains non-alphanumeric characters, not a woogles game ID
+                return NULL;
+            }
+        }
+        // It's alphanumeric and not all digits, assume it's a woogles game ID
+        game_id_str = string_duplicate(identifier);
+    }
+    
+    char *gcg_content = NULL;
+
+    if (game_id_str) {
+        // Use woogles API to get GCG content
+        char *curl_cmd = get_formatted_string("curl -s -H 'Content-Type: application/json' "
+                                              "https://woogles.io/api/game_service.GameMetadataService/GetGCG "
+                                              "-d '{\"game_id\":\"%s\"}'", game_id_str);
+        
+        char *response = get_process_output(curl_cmd);
+        free(curl_cmd);
+        free(game_id_str);
+        
+        if (response) {
+            // Parse JSON response to extract GCG content
+            const char *gcg_start = strstr(response, "\"gcg\":\"");
+            if (gcg_start) {
+                gcg_start += 7; // Skip "\"gcg\":\""
```

Comment:
Instead of "7" prefer to assign "\"gcg\":\"" to use a define statement at the top of the file to use above like  strstr(response, WOOGLES_GCG_START); and use gcg_start += string_length(WOOGLES_GCG_START);

That way if you need to change the string, you don't have to remember to edit two places.